### ISSUE TITLE
Repository-OSGenomics-phs000699.v1.p1-20161129-KL

### ISSUE THREAD

**Issue by adurbin2** [UCSC-Treehouse/operations#13]
Setting up the first of four RNASeq batches for @cygao to RNASeq. This one is for Katrina's dataset (Osteosarcoma Genomics), n=35 samples. (Please process each batch separately to avoid confusion.) 

Manifest will be uploaded soon. 

Please put the output folders here:
/pod/pstore/groups/treehouse/backed_up/archive/output/externalSamples/phs000699.v1.p1_OSGenomics

**Comment by klearned** [comment]
Here is the manifest.
[phs000699RnaSeqManifest.txt](https://github.com/UCSC-Treehouse/operations/files/623406/phs000699RnaSeqManifest.txt)

**Comment by rcurrie** [comment]
Only run RNASeq + QC, no fusion

**Comment by rcurrie** [comment]
@klearned the manifest attached only had 24 samples listed (24 fastq pairs). I'm going to start it running anyways and if you have a new one with just added samples I can run those after (ok to just add into a final manifest as treeshop will skip over ones it has already run as long as the names don't change). Also the format of these manifests isn't correct. They should be tsv files with a header row exactly as the core requires and the file names should end in a '/'. I've cleaned this one up in gdocs:

https://docs.google.com/a/ucsc.edu/spreadsheets/d/1no8jYVwTqby29lqrJOYlHvx4TEvuo7jBoFVfQoiAXjs/edit?usp=sharing

We should probably switch to just placing them in the destination folder so they are all kept together. Also will be useful when we start including the other meta data columns on the way to the core processing these.

**Comment by rcurrie** [comment]
These are now running on rob's treeshop

**Comment by adurbin2** [comment]
@rcurrie There is a certain beauty in putting the input files and the output files into the same folder, and I think it's a valid possibility ... however that is generally not the way it has been done up to now.  We might also consider putting other types of data into the same folder (QC for example). Perhaps we can pause briefly for a review/overhaul of Pod storage principles, so that you and Holly and Katrina and I and others on TH can all contribute views, & get ourselves onto the same page?

**Comment by klearned** [comment]
@rcurrie Thanks for letting me know about some files being missing, and I am sorry about the incorrect manifest format. I missed some lines when copying the manifest I created on /pod into the .txt file (so I could attach the file to the ticket), I forgot we were going to add in the core headings, and I think the tabs became spaces when I copied the file into the .txt file. Next time, I'll just point to the manifest file on /pod instead of trying to attach it to the ticket so there will hopefully be fewer errors!

Here is the revised tsv manifest with the headers (just the two headers, as you did in your spreadhseet) and all the 35 samples (I also went ahead and added the missing lines to your spreadsheet - which I noticed had a trailing "}" at the end of the last filename of the last sample listed, SRR1701598 - I removed the trailing "}"):
`/pod/pstore/groups/treehouse/published-data/dbGap/phs000699.v1.p1/phs000699RnaSeqManifest`

Going forward, do you want the manifests to have all the Core manifest heading fields or just the two we're using currently (as you have done in your revised manifest) as the first line of the manifest?

I don't know what you mean by "...and the file names should end in a '/' ." I don't see that in your cleaned up manifest spreadsheet so I think I must be misunderstanding.

When you say we should switch to just placing "them" in the destination folder, do you mean putting the manifest file in the destination folder, or the input fastq files in the destination folder? I read it as the manifest file, but I think @adurbin2 is reading it the input fastq files. Thanks!

**Comment by rcurrie** [comment]
4:39PM Update: 3 files complete and unfortunately they all failed:
./SRR1701090/qc/readDist.txt_FAIL_qc.txt
./SRR1701117/qc/readDist.txt_FAIL_qc.txt
./SRR1701102/qc/readDist.txt_FAIL_qc.txt

4 more are processing in parallel. Will let initial manifest run to completion and then run the updated manifest - treeshop will automatically just process the added ones. At this rate its likely going to take most of this week. Will send updates as I see pass/fail.

**Comment by rcurrie** [comment]
Wrt Katrina's questions:

The only fields treeshope uses are "Submitter Sample ID" and "File Path". You can have others in there as well and it will just ignore them. My thinking is we can start putting more of the data that the core will store in these though. That way we can down the road ingest them all into the new storage system including meta data.

My bad "should NOT end in a /", sorry for the confusion.

I think put the manifest file in the destination. That was as above we'd have the manifest, outputs, and methods all in the same location. We could also put fastq's in there as well I suppose so its all in one place and therefore easy to move to the new storage system as well as implement data retention policies (when to delete bams...). I'm nominally thinking we take up that whole subject Jan 2nd :-)

**Comment by klearned** [comment]
@rcurrie Thank you for the update.

@hbeale and @rcurrie 
Honestly, I am freaking out a bit that so much data that I have been spending a gazillion hours on (to find, research terms of use, apply for permission, figure out how to download, download, convert into fastqs, and find clinical data for) is failing. It's more than a little frustrating at this point. Sorry for the rant, I know it's not anyone's fault, it just feels so disappointing!

Looking at the QC output for the first sample, I see:
input   uniqMappedNonDupeReadCount      estExonicUniqMappedNonDupeReadCount     qc
/data/readDist.txt      3542012.5       3270333.6       FAIL

So this means that there are 3.2 million uniquely mapped, non-duplicate exonic reads, correct? And, the threshold for our QC is 10 million uniquely mapped, non-duplicate exonic reads, so that is why this data is failing?

Can the data submitters be using this data reliably at this quality? I am wondering if this could indicate a problem with the submission of data by the submitter?

**Comment by rcurrie** [comment]
@hbeale it appears treeshop has processed 14 of these samples as of this morning but there is only 4 of them where a pass/fail file has been generated. For example under:

/pod/pstore/groups/treehouse/backed_up/archive/output/externalSamples/phs000699.v1.p1_OSGenomics

SRR1701090 has all the qc output I'd expect (pdf, etc...) but SRR1701742 only has 4. 

readDist.txt
rnaAligned.sortedByCoord.md.bam
rnaAligned.out.md.sorted.geneBodyCoverage.txt
rnaAligned.sortedByCoord.md.bam.bai

The treeshop script writes out the methods.json file last so I don't think its that its failing completely. And the copy of the qc folder is a single operation (fabric get). Also many of the others only have these 4 as well.

**Comment by hbeale** [comment]
@rcurrie 
I've looked at the QC output. When there are only 4 files they are usually very small (like these below), and it's always been because the process failed. I looked into the methods.json but didn't see anything for troubleshooting. Do you have the error messages from the QC docker?


-rw-r--r-- 1 rcurrie pawg  303 Dec  7 03:44 rnaAligned.out.md.sorted.geneBodyCoverage.txt
drwxr-xr-x 2 rcurrie pawg 2.0K Dec  7 03:44 .
-rw-r--r-- 1 rcurrie pawg    0 Dec  7 03:44 readDist.txt
-rw-r--r-- 1 rcurrie pawg  148 Dec  7 03:44 rnaAligned.sortedByCoord.md.bam
-rw-r--r-- 1 rcurrie pawg   16 Dec  7 03:44 rnaAligned.sortedByCoord.md.bam.bai

**Comment by rcurrie** [comment]
methods just prints what was run and on what, no errors

Unfortunately I don’t have an error log partially because the output of the pipeline is mostly status bars:

[13.93.226.157] out: [===========================================================================>  ]
[13.93.226.157] out: [===========================================================================>  ]
[13.93.226.157] out: [===========================================================================>  ]
[13.93.226.157] out: [===========================================================================>  ]
[13.93.226.157] out: [===========================================================================>  ]              [13.93.226.157] out: [===========================================================================>  ]
[13.93.226.157] out: [===========================================================================>  ]
[13.93.226.157] out: [===========================================================================>  ]              [1

or 

[13.93.226.157] out:    61300 transcripts finished
[13.93.226.157] out:    61400 transcripts finished
[13.93.226.157] out:    61500 transcripts finished
[13.93.226.157] out:    61600 transcripts finished
[13.93.226.157] out:    61700 transcripts finished                                                                 [13.93.226.157] out:    61800 transcripts finished
[13.93.226.157] out:    61900 transcripts finished

Suggest running on just one of the ones that manually failed on the command line.

Rob

**Comment by hbeale** [comment]
Where would I find the bam that was generated by the RNA-Seq pipeline?

I don't see one here:
/pod/pstore/groups/treehouse/backed_up/archive/output/externalSamples/phs000699.v1.p1_OSGenomics/SRR1701343

or here
/pod/pstore/groups/treehouse/backed_up/archive/output/externalSamples/bams/phs000699

**Comment by klearned** [comment]
These never had bam files. When I download them from dbGap/SRA, they come down as .sra files and then I converted them to fastq files. The .sra files are here:

`/pod/pstore/groups/treehouse/published-data/dbGap/phs000699.v1.p1/data/sra`

**Comment by klearned** [comment]
oh, never mind!! I misread your comment! Ignore my last comment!

**Comment by rcurrie** [comment]
Under the qc folder - the one rnaseq generates is deleted/not saved as we agreed that the one qc generates was the more useful one to archive.

**Comment by hbeale** [comment]
I've looked at a ton of things and the only pattern I saw was the the samples where QC succeeded generally had smaller input. 

inputR1name; inputR1bytes; qcGenerated 588392354 SRR1701117_1.fastq.gz succeeded  
inputR1name; inputR1bytes; qcGenerated 1196309192 SRR1701090_1.fastq.gz succeeded  
inputR1name; inputR1bytes; qcGenerated 2886460161 SRR1701307_1.fastq.gz succeeded  
inputR1name; inputR1bytes; qcGenerated 3149022133 SRR1701102_1.fastq.gz succeeded  
inputR1name; inputR1bytes; qcGenerated 3882304566 SRR1701572_1.fastq.gz failed  
inputR1name; inputR1bytes; qcGenerated 3947148014 SRR1701153_1.fastq.gz failed  
inputR1name; inputR1bytes; qcGenerated 4056166592 SRR1701742_1.fastq.gz failed  
inputR1name; inputR1bytes; qcGenerated 4074811396 SRR1701812_1.fastq.gz failed  
inputR1name; inputR1bytes; qcGenerated 4149395479 SRR1701662_1.fastq.gz succeeded  
inputR1name; inputR1bytes; qcGenerated 4320371450 SRR1701172_1.fastq.gz failed  
inputR1name; inputR1bytes; qcGenerated 4323470837 SRR1701197_1.fastq.gz failed  
inputR1name; inputR1bytes; qcGenerated 4326618404 SRR1701291_1.fastq.gz succeeded  
inputR1name; inputR1bytes; qcGenerated 4582870491 SRR1701622_1.fastq.gz failed  
inputR1name; inputR1bytes; qcGenerated 4809164083 SRR1701796_1.fastq.gz failed  
inputR1name; inputR1bytes; qcGenerated 4869064495 SRR1701405_1.fastq.gz failed  
inputR1name; inputR1bytes; qcGenerated 5185968878 SRR1701343_1.fastq.gz failed  
inputR1name; inputR1bytes; qcGenerated 5352685341 SRR1701511_1.fastq.gz failed  


commands:
sampleStatus="SRR1701090+succeeded
SRR1701102+succeeded
SRR1701117+succeeded
SRR1701291+succeeded
SRR1701307+succeeded
SRR1701662+succeeded
SRR1701153+failed
SRR1701172+failed
SRR1701197+failed
SRR1701343+failed
SRR1701405+failed
SRR1701511+failed
SRR1701572+failed
SRR1701622+failed
SRR1701742+failed
SRR1701796+failed
SRR1701812+failed"

cd /pod/pstore/groups/treehouse/published-data/dbGap/phs000699.v1.p1/data/fastq/
for ss in $sampleStatus; do sample=${ss/+*}; status=${ss/*+}; 
echo "inputR1name; inputR1bytes; qcGenerated" `ls -l ${sample}_1.fastq.gz | cut -f5,9 -d" "`   $status " "  
done | sort -k4 -n

**Comment by rcurrie** [comment]
As soon as the first set of this batch (the manifest originally only had part of the data), I'll pause and re-run one of those that failed using treeshop to both try and capture a log as well as rule out that its something about the treeshop wrapper. Hopefully we can then sort and then I'll finish up the balance of this batch (~15)

**Comment by klearned** [comment]
Thanks @hbeale and @rcurrie for your careful work on this!

**Comment by hbeale** [comment]
@rcurrie sounds like a good plan. If it's easy to keep the bam output from the rna-seq-cgl pipeline, that would be great too so i don't have to restart troubleshooting at fastq

**Comment by rcurrie** [comment]
I’ll turn that ‘prune’ off for the next run.

Rob

**Comment by hbeale** [comment]
Thank you!

**Comment by rcurrie** [comment]
First batch finished: 23 have rnaseq output. Of those 4 PASSED and 4 FAILED. The balance didn't generate a pass/fail from qc. I'm going to run one of those with logging to sort out what went wrong with holly, and once we have it sorted will start this treeshop back up to process the balance (35-23=12)

**Comment by klearned** [comment]
@rcurrie Thank you for the update

**Comment by rcurrie** [comment]
Started processing which failed to generate all the qc output. Unfortunately the initial run was pruning so I don't have the rnaseq bam so will have to run rnaseq first and then qc. I think I've sorted how to save the log (issues due to running parallel). Will notify when I a full log.

**Comment by hbeale** [comment]
Thanks, Rob.

**Comment by rcurrie** [comment]
TL;DR; Appears rnaseq is generating a zero size bam file, how rude!

Good news: Job ran on SRR1701812 (which failed in the first run to generate QC PASS/FAIL)
Bad news: Had a bug in the new prune option so QC didn't run
Good news: Exited before deleting so the SRR1701812.sorted.bam was still on the machine, and lo and behold its zero length!

@hbeale the log of the run is at:
/pod/pstore/groups/treehouse/backed_up/archive/output/externalSamples/phs000699.v1.p1_OSGenomics/debug/log.txt

I don't see anything out of the ordinary wrt the rnaseq run. Maybe take a look at the rnaseq output:

/pod/pstore/groups/treehouse/backed_up/archive/output/externalSamples/phs000699.v1.p1_OSGenomics/debug/SRR1701812/rnaseq

and let me know if it seems clearly wrong.

@cygao I think in general we have not been backhauling the bams from our pipeline0/1/2 days, but if we have were there any where the bam was empty that your remember but rnaseq still outputed results? If not then maybe we run this sample on pipeline0 just to see if it generates a bam as a debugging measure of the rnaseq pipeline. I've looked through the fabfile and can't see anywhere that I might be touching the bam output of rnaseq.

**Comment by hbeale** [comment]
In many of Jeltje's runs, the bam file had a EOF error. Although I don't
recall any being zero length, the EOF error was common. She had to rerun
the pipeline to get a fixed bam file.

I looked at the output from the samples that previously failed to run QC
and nothing seemed wrong or different in comparison to the ones that were
able to run QC. I'm afraid this is an engineering question. Science
context: Accurate RSEM and Kallisto quantification doesn't require the
genome-aligned bam file; I assume it does require the transcriptome-aligned
bam, but we can't run QC from that bam file.

**Comment by rcurrie** [comment]
Sounds like there may be a bug somewhere in the rnaseq pipeline that's been there for a while but ignored as we didn't use the bam output. In the interest of keeping the train running as per isabelle @hbeale I'm going to crank back up this pipeline to process the rest of the samples in the batch (35-23=12) and will backhaul the bam so we can inspect all of them. Those plus the logs hopefully will provide JV some hints if they are all zero.

**Comment by hbeale** [comment]
Sounds good

**Comment by rcurrie** [comment]
Of 35:
30 processed, 5 remaining
6 PASS, 4 FAIL,  20 no QC output (will examine log and bam from these running to sort)

@klearned @adurbin2 to get a PASS/FAIL count, from within the output directory of a treeshop run:

echo PASS && find . -name "*PASS*" | wc && echo FAIL && find . -name "*FAIL*" | wc                               PASS                                                                                                                     
6       6     246
FAIL
4       4     164

**Comment by rcurrie** [comment]
All 35 samples have been processed:

rsem.genes.norm_counts.hugo.tab was generated for all of them and ~900k for each so it appears

PASS: 7 FAIL: 4 => 24 failed to generate qc pass fail

I backhauled the <samplename>.sorted.bam from this second batch of 12 - 3 of them had data and 9 were zero length:

   0 ./SRR1701137/SRR1701137.sorted.bam     0 ./SRR1701598/SRR1701598.sorted.bam
   0 ./SRR1701216/SRR1701216.sorted.bam     0 ./SRR1701638/SRR1701638.sorted.bam
 26G ./SRR1701256/SRR1701256.sorted.bam     0 ./SRR1701687/SRR1701687.sorted.bam
   0 ./SRR1701272/SRR1701272.sorted.bam  8.6G ./SRR1701708/SRR1701708.sorted.bam
   0 ./SRR1701369/SRR1701369.sorted.bam     0 ./SRR1701777/SRR1701777.sorted.bam
   0 ./SRR1701489/SRR1701489.sorted.bam   17G ./SRR1701828/SRR1701828.sorted.bam

@hbeale sure looks like some sort of engineering issue in RNASeq, if you agree will file a separate issue against JV with a pointer to the inputs for all of theses above.

**Comment by hbeale** [comment]
Hi Rob -- 

I agree that the problem is malformed or unformed bams. 

When non-zero sized bam files generated by the rna-seq-cgl pipeline cannot be analyzed, they usually have truncated input

For future ref, if you have access to samtools you can test whether the bam files with content are correctly formatted with this command:

`for i in *.bam ; do echo -n $i; samtools view -h $i | head -1 ; done 2>&1 | grep '\['`

aka https://gist.github.com/hbeale/16429cd113b45c1d85574c349768ac44

**Comment by rcurrie** [comment]
Filed as:

https://github.com/BD2KGenomics/cgl-docker-lib/issues/215

**Comment by rcurrie** [comment]
Passing to @e-t-k : Please import TPM values for the 31 that did not fail into MedBook for addition to Compendium V3

**Comment by e-t-k** [comment]
The following 4 samples will be excluded, based on the presence of readDist.txt_FAIL_qc.txt files in their /qc/ dir:

SRR1701090
SRR1701117
SRR1701102
SRR1701307

The remaining 31 samples, listed below, will be imported as follows:
* data is the TPM column from RSEM/rsem_genes.results
* the ensembl IDs from this file will be translated to Hugo gene names
* log2(x+1) normalization **will** be applied during the input
* Study: "**OSGenomics phs000699.v1.p1 2016/11/29**"
* Data Set: "**OSGenomics phs000699.v1.p1 2016/11/29 (TPM)**"

SRR1701137 SRR1701153 SRR1701172 SRR1701197 SRR1701216 SRR1701256 SRR1701272 SRR1701291 SRR1701327 SRR1701343 SRR1701369 SRR1701405 SRR1701435 SRR1701454 SRR1701473 SRR1701489 SRR1701511 SRR1701527 SRR1701556 SRR1701572 SRR1701598 SRR1701622 SRR1701638 SRR1701662 SRR1701687 SRR1701708 SRR1701742 SRR1701777 SRR1701796 SRR1701812 SRR1701828

**Comment by klearned** [comment]
Paper: [PMID 25512523](http://www.ncbi.nlm.nih.gov/pubmed/25512523)

**Comment by e-t-k** [comment]
The 31 samples [listed above](https://github.com/UCSC-Treehouse/operations/issues/13#issuecomment-266858155) have been imported to MedBook **under the TPM normalization, not norm_counts**.

The 4 "Fail" samples remain un-imported.

**Comment by adurbin2** [comment]
Hi Rob @rcurrie Just for the record, what version of RNASeq was this batch run on?

**Comment by rcurrie** [comment]
2.0.8

In each result is a methods.json which specifies the versions of each of the components.

**Comment by adurbin2** [comment]
Good to know - thanks!

**Comment by klearned** [comment]
@adurbin2 @rcurrie @e-t-k 
This dataset has been added to the V3 Compendium and imported to Medbook, and possibly had fusion run on it to (see [this comment above](https://github.com/UCSC-Treehouse/operations/issues/13#issuecomment-271020255)). Is this waiting for another step or should we close this one?

**Comment by e-t-k** [comment]
@adurbin2 I'm fine with closing as long as we don't need to do anything with the 4 QC failed samples. Thank you!

**Comment by klearned** [comment]
Good point! Maybe we should take another look at the failed files?

**Comment by klearned** [comment]
I looked at the 4 failed samples:
SRR1701090
SRR1701117
SRR1701102
SRR1701307
...and found that they all failed because there aren't enough uniquely mapping non-duplicate reads (uniqMappedNonDupeReadCount). To pass, they must have >10,000,000 and all of these fell under that threshold. At this point, I don't think there is anything more to do with these four. We threw around the idea of having a back-up compendium comprised of data that didn't make the cut of having 10 million uniquely mapping non-duplicate reads. We could compare a prospective to this B-team compendium as a last resort. Nothing has come of that at this point, however. So, I am going to go ahead and close this ticket.
